scd30: Allow setting ambient pressure compensation

matoxp opened the pull request:
## Description:

This PR allows configuring the ambient pressure compensation for the Sensirion SCD30 CO2 sensor.

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#842

## Checklist:
  - [x ] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

matoxp commented:
I missed required changes in sensor.py and now amended my initial commit. However, this doesn't seem to trigger another round of checks. Shall I close this PR and resubmit?

jesserockz commented:
Hi @matoxp If you feel up to it, do you want to add the config validations for pressure as well so the config entry could be:
```
ambient_pressure_compensation: 961mBar
```

It would be very similar to the validations around  https://github.com/esphome/esphome/blob/dev/esphome/config_validation.py#L614-L645

It could just start with one single one to use `bar` which will do the unit conversions automatically and we could add other units later on similar to `temperature`

matoxp commented:
Hi @jesserockz  based on your suggestion, I added a new config validation for pressure (supporting just mBar for now) and also rebased against latest dev. I will update my PR for esphome-docs accordingly.

jesserockz commented on esphome/config_validation.py:
```diff
@@ -622,6 +622,7 @@ def validator(value):
 _temperature_k = float_with_unit("temperature", "(° K|° K|K)?")
 _temperature_f = float_with_unit("temperature", "(°F|° F|F)?")
 decibel = float_with_unit("decibel", "(dB|dBm|db|dbm)", optional_unit=True)
+pressure = float_with_unit("pressure", "(mBar)", optional_unit=True)
```
If you add it as just `Bar` the `float_with_unit` allows pretending the `m` for milli

matoxp committed:
scd30: Allow setting ambient pressure compensation

bobobo1618 commented:
Apologies if this is the wrong place to discuss it but does ESPHome already have some mechanism in place through which data from one sensor can be used to feed another?

My current (custom) SCD30 implementation uses a separate pressure sensor to feed the SCD30's altitude/pressure compensation and I'd like to know if it's possible to port it to ESPHome.

jesserockz commented:
@bobobo1618 Please open a new [feature-request](https://github.com/esphome/feature-requests/issues/new/choose) for this. In short it would be possible with some minor changes
